packages/front-end/components/SlackIntegrations/SlackIntegrationAddEditModal/SlackIntegrationAddEditModal.tsx

```diff
@@ -0,0 +1,273 @@
+        {...form.register("name")}
+        onChange={(evt) => {
+          form.setValue("name", evt.target.value);
+          handleFormValidation();
+        }}
+      />
+
+      <Field
+        label="Description"
+        placeholder="(optional description)"
+        autoComplete="off"
+        {...form.register("description")}
+        onChange={(evt) => {
+          form.setValue("description", evt.target.value);
+          handleFormValidation();
+        }}
+      />
+
+      <div className="form-group">
+        <label className="d-block">Event filters</label>
+        <div className="mt-1">
+          <Typeahead
+            id="events-input"
+            labelKey="name"
+            multiple={true}
+            allowNew={false}
+            options={eventWebHookEventOptions.map(({ id }) => {
+              return {
+                id: id,
+                name: id,
+              };
+            })}
+            onChange={(
+              selected: {
+                id: NotificationEventName;
+                name: NotificationEventName;
+              }[]
+            ) => {
+              form.setValue(
+                "events",
+                selected.map((item) => item.id)
+              );
+              handleFormValidation();
+            }}
+            selected={form.watch("events").map((v) => ({ id: v, name: v }))}
+            placeholder="Choose events"
+            positionFixed={true}
+          />
+          <small className="text-muted">
+            Only receive notifications for matching events.
+          </small>
+        </div>
+      </div>
+
+      <Field
+        label="Slack App ID"
+        autoComplete="off"
+        helpText="Copy the Slack App ID from the app's Basic Information page"
+        required
+        {...form.register("slackAppId")}
+        onChange={(evt) => {
+          form.setValue("slackAppId", evt.target.value);
+          handleFormValidation();
+        }}
+      />
+
+      <Field
+        label="Slack App Incoming Webhook URL"
+        autoComplete="off"
+        helpText="Copy the Incoming Webhook URL for your Slack App. This can be found on the Incoming Webhooks page under Features for your Slack app configuration"
+        required
+        {...form.register("slackIncomingWebHook")}
+        onChange={(evt) => {
+          form.setValue("slackIncomingWebHook", evt.target.value);
+          handleFormValidation();
+        }}
+      />
+
+      <Field
+        label="Slack Signing Key"
+        autoComplete="off"
+        required
+        helpText="Copy the signing key from the app's Basic Information page"
+        {...form.register("slackSigningKey")}
+        onChange={(evt) => {
+          form.setValue("slackSigningKey", evt.target.value);
+          handleFormValidation();
+        }}
+      />
+
+      <div className="form-group">
+        <label className="d-block">Environment filters</label>
+        <div className="mt-1">
+          <MultiSelectField
```

Comment:
`MultiSelectField` has a `label` prop you can use that has the same styling as `<Field>`.  It will let you get rid of this wrapping `form-group` div as well.

Reply:
Awesome. I've made that change. I've also pushed up a change to replace the TypeAhead component in the Event Webhooks and Slack integration add/edit modals.